### PR TITLE
Introduce FEATURE_GDBJIT_FRAME feature

### PR DESCRIPTION
**Pull request by parjong**
This commit introduce a new feature called FEATURE_GDBJIT_FRAME.

When it is turned on along with FEATURE_GDBJIT, CLR always emits .debug_frame section and notifies GDB for all JITed methods.

### CODE REVIEW

**Comment by parjong** [comment]
@chunseoklee @lucenticus Please take a look.

**Comment by parjong** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test please

**Comment by BruceForstall** [comment]
@janvorli

**Comment by lucenticus** [comment]
CC: @ayuckhulk

File: src/vm/gdbjit.cpp [hunk 1]
```diff
@@ -1848,6 +1808,590 @@ static inline bool isListedModule(const WCHAR *wszModuleFile)
...
+#if defined(_TARGET_ARM_)
+    const unsigned int code_alignment_factor = 2;
+    const int data_alignment_factor = -4;
+
+    UINT8 cieCode[] = {
+        // DW_CFA_def_cfa 13[sp], 0
+        0x0c, 0x0d, 0x00,
+    };
+
+    UINT8 fdeCode[] = {
```
Comment (ayuckhulk) [inline src/vm/gdbjit.cpp]: Does this code need to be in sync with JIT compiler?

Comment (parjong) [inline src/vm/gdbjit.cpp]: Yes, this code should be synced with JIT. Unfortunately, there is no easy way.

I first tried to use GCInfo, but it turns out that GCinfo is not suitable for that purpose. GCInfo lacks several necessary information such as which registers are stored, and where they are.

I guess that unwind info may provide all the information needed here, but unwind info is a bit complicated.

So, this commit tries to provide the same frame information for all the managed methods. Even though that frame information is incomplete (and sometimes wrong), but this allows ``gdb`` to unwind some managed stack frames (``gdb`` currently could not unwind any managed stack frame on ARM).

Comment (parjong) [inline src/vm/gdbjit.cpp]: @ayuckhulk Please take a look

Comment (ayuckhulk) [inline src/vm/gdbjit.cpp]: The changes look good to me and they do not break our internal debugger tests.

**Comment by parjong** [comment]
@jkotas @janvorli Could you please merge this PR?

**Comment by parjong** [comment]
@janvorli Thank you!

**Pull request merged**